CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Developing the code
+
+An appropriate environment for developing a package can be entered using the `env` attribute of the package. For example:
+```
+nix-shell default.nix -A language-plutus-core.env
+```
+
+There are also `shell.nix` files in the package subdirectories.
+
+## Updating the generated Haskell package set
+
+`pkgs/default.nix` contains a generated package set with all the dependencies for this project.
+
+You should regenerate this if you change any dependencies in cabal files. To do this, use the environment defined in `shell.nix`, and run `pkgs/generate.sh`.
+
+## Code style
+
+We use `stylish-haskell` and `hlint`, and enable a large number of GHC warnings.
+- These are run by the CI, so if you don’t use them your PR will not go green. To avoid annoyance, set up your editor to run them automatically.
+- It’s fine to be aggressive about disabling `hlint` rules that don’t provide value, but check with the team before doing this. Err on the side of disabling them globally rather than for specific files, so we stay consistent.
+- Try and make sure that you do not have any compiler warnings - this is not currently checked by the CI, but may be in future.
```

Comment:
Perhaps this should include a line on cabal files (`default-extensions`, `other-extensions`, `ghc-options`, and whether or not `-any` should be added to dependencies without bounds)